packages/commerce-sdk-react/src/test-utils.tsx

```diff
@@ -135,3 +140,31 @@ export const assertInvalidateQuery = (
 export const assertRemoveQuery = (queryResult: UseQueryResult) => {
     expect(queryResult.data).not.toBeDefined()
 }
+
+const getQueryName = (method: string): string => {
+    // Most query endpoints start with 'get'; Shopper Login retrieveCredQualityUserInfo also exists
+    const prefix = /^get|^retrieve/
+    // If it exists, replace the prefix verb with 'use'
+    if (prefix.test(method)) return method.replace(prefix, 'use')
+    // Otherwise just prefix the method with 'use' and fix the case
+    return method.replace(/^./, (ltr) => `use${ltr.toUpperCase()}`)
+}
+
+export const expectAllEndpointsHaveHooks = (
```

Comment:
Should we include list if not_implemented list here to use for other hook rather than just basket?

Reply:
So this might not be the most intuitive thing, but I changed how we detect "not implemented", and this test doesn't quite check that. We don't have a "not implemented" list any more. Instead, "implemented" means "_value_ is not `undefined` in the `cacheUpdateMatrix`". In addition to that, I made a `TODO` helper that logs "this method is not implemented" and returns `undefined`. Then, in the `cacheUpdateMatrix`, we put `[method]: TODO(method)`. This means that the method is still "not implemented", but it's more explicitly visible because it's explicitly present in the `cacheUpdateMatrix` (and logs a warning to console).

This test doesn't quite check for "not implemented". It only checks the _keys_ of the `cacheUpdateMatrix`, rather than the _values_. So, to this test, every `TODO` is considered implemented. That way we never need an explicit list of "not implemented". But, writing this out, I feel like the keys vs values distinction is a bit too subtle, so I'll change the test to check values and have an explicit list of "not implemented" in the assertion.